design/ceph/object/swift-and-keystone-integration.md

```diff
@@ -0,0 +1,312 @@
+---
+title: swift-and-keystone
+target-version: release-X.X
+---
+
+# Swift and Keystone Integration
+
+## Summary
+
+### Goals
+
+The goal of this proposal is to allow configuring the Swift API and
+the Keystone integration of Ceph RGW natively via the Object Store
+CRD, which allows native integration of the Rook operated Ceph RGW into
+OpenStack Clouds.
+
+Both changes are bundled together as one proposal because they will
+typically deployed together. It is unlikely to encounter a Swift
+object store API outside an OpenStack cloud and in the context of an
+OpenStack cloud the Swift API needs to be integrated with the Keystone
+authentication service.
+
+It must be possible to serve S3 and Swift for the same object store
+pool, e.g. by serving S3 and Swift from the same RGW instance.
+
+Any changes to the CRD must be future safe and cleanly allow extension
+to further technologies (such as LDAP authentication).
+
+### Non-Goals
+
+* Support for Keystone API versions below v3. API version v2 has long
```

Comment:
So Keystone API v3 or newer is the goal? Shall we mention that in the goals section?

Reply:
This is explicitly stated now.